Tests/test_death_handling.gd

```diff
@@ -0,0 +1,172 @@
+extends GutTest
```

Comment:
after pulling the branch, I realized that this test doesn't use the structure that was made in #63 
Normal, since this PR was started first, but could you update this to extend from `TestBase` ? You will be able to remove all variable declarations, most of the `before_each` (except after the `watch_signals`, but you can call `super` for everything before)
You will need to leave the the `after_each`, and could you also add an `assert_no_new_orphans` as in done in other tests ? This is to check for runtime errors mainly (cf #51 and #62 )

Reply:
Done